climpred/prediction.py

```diff
@@ -12,28 +11,82 @@
     PM_COMPARISONS,
     PROBABILISTIC_HINDCAST_COMPARISONS,
     PROBABILISTIC_PM_COMPARISONS,
-    __e2c,
 )
-from .constants import CLIMPRED_DIMS, M2M_MEMBER_DIM, PM_CALENDAR_STR
+from .constants import CLIMPRED_DIMS, CONCAT_KWARGS, M2M_MEMBER_DIM, PM_CALENDAR_STR
 from .logging import log_compute_hindcast_header, log_compute_hindcast_inits_and_verifs
-from .metrics import (
-    DETERMINISTIC_HINDCAST_METRICS,
-    HINDCAST_METRICS,
-    METRIC_ALIASES,
-    PM_METRICS,
-)
+from .metrics import HINDCAST_METRICS, METRIC_ALIASES, PM_METRICS
+from .reference import historical, persistence
 from .utils import (
     assign_attrs,
     convert_time_index,
     copy_coords_from_to,
     get_comparison_class,
-    get_lead_cftime_shift_args,
     get_metric_class,
-    shift_cftime_index,
 )
 
 
-def get_metric_comparison_dim(metric, comparison, dim, kind):
+def _apply_metric_at_given_lead(
```

Comment:
I envision this being much neater and nicer with class inheritance. It will be under `Scoring()`.